test/metabase/query_processor_test/cumulative_aggregation_test.clj

```diff
@@ -255,30 +275,36 @@
                   [#t "2017-02-01" #t "2017-01-01" 22338.43]
                   [#t "2017-03-01" #t "2017-01-01" 36454.11]]
-                 (mt/formatted-rows [->local-date ->local-date 2.0]
-                   (qp/process-query query)))))))))
+                 (mt/formatted-rows [->local-date ->local-date 2.0] (qp/process-query query)))))))))
 
 (deftest ^:parallel cumulative-sum-with-three-breakouts-test
   (testing "Three breakouts: should be ORDERED BY first BREAKOUT and PARTITIONED BY last two BREAKOUTS (#2862, #42003)"
     (mt/test-drivers (mt/normal-drivers)
-      (let [query (-> (mt/mbql-query orders
-                        {:aggregation [[:cum-sum $total]]
-                         :breakout    [!day.created_at !year.created_at !month.created_at]
-                         :limit       4})
+      (let [query (-> (lib/query
+                       (lib.tu/merged-mock-metadata-provider
+                        (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+                        {:settings {:report-timezone "UTC"}})
+                       (mt/mbql-query orders
+                         {:aggregation [[:cum-sum $total]]
+                          :breakout    [!day.created_at !year.created_at !month.created_at]
+                          :limit       4}))
                       (assoc-in [:middleware :format-rows?] false))]
         (mt/with-native-query-testing-context query
           ;; you can sanity check these numbers by changing `:cum-sum` to `:sum` and adding them up manually.
           (is (= [[#t "2016-04-30" #t "2016-01-01" #t "2016-04-01" 52.76]
-                  [#t "2016-05-04" #t "2016-01-01" #t "2016-05-01" 98.78]  ; <-- total should reset here, when last two breakouts change
+                  [#t "2016-05-04" #t "2016-01-01" #t "2016-05-01" 98.78] ; <-- total should reset here, when last two breakouts change
                   [#t "2016-05-06" #t "2016-01-01" #t "2016-05-01" 186.07]
                   [#t "2016-05-08" #t "2016-01-01" #t "2016-05-01" 270.94]]
-                 (mt/formatted-rows [->local-date ->local-date ->local-date 2.0]
-                   (qp/process-query query)))))))))
+                 (mt/formatted-rows
+                  [->local-date ->local-date ->local-date 2.0]
+                  (qp/process-query query)))))))))
 
 (deftest ^:parallel cumulative-count-and-sum-in-expressions-test
   (testing "Cumulative count should work inside expressions (#13634, #15118)"
     (mt/test-drivers (mt/normal-drivers-with-feature :window-functions/cumulative)
-      (let [metadata-provider (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+      (let [metadata-provider (lib.tu/merged-mock-metadata-provider
+                               (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+                               {:settings {:report-timezone "UTC"}})
```

Comment:
So for the most part these actually did work on CI and did work locally unless you did something like `(metabase.driver/report-timezone! "US/Pacific")` and then tried to run tests in your REPL. There was kind of this implicit assumption in a lot of tests that `report-timezone` was either unset or set to `UTC` in order for tests to pass. 

So it's a little unrelated to the core changes in the PR but I tried to go in and fix a bunch of tests that would fail in the REPL if you changed your `report-timezone` before running them.

We could have done this with `with-temporary-setting-values` everywhere (which would affect the app DB metadata provider) but since I haven't made it thread-safe **yet** I did it this way instead so I could keep these tests `^:parallel`.

I did think about adding a helper like `lib.tu/metadata-provider-in-timezone` but I thought `"UTC"` vs `{:settings {:report-timezone "UTC"}}` as the argument really isn't that much shorter and it's making the behavior less obvious, with `merged-mock-metadata-provider` it's pretty clear I think that you're overriding the `report-timezone` Setting which people are already familiar with whereas `metadata-provider-in-timezone` seemed like more of a mystery, if you saw that you might not know what it's doing unless you went in and looked at its definition.

Also `merged-mock-metadata-provider` is flexible if we need to add more stuff to it like `:start-of-week` or other stuff

Reply:
I agree tho that it is too easy to forget to set `report-timezone` when it's relevant to the test results. I'm trying to think what the best way to address this is going forward. I think once I do a follow-on PR to make `with-temporary-setting-values` thread safe then we can just add test fixtures that set `report-timezone` to `"UTC"` and `start-of-week` to `:sunday` in the namespaces that have tests where the value of these settings matter, then you don't need to do it in every individual test (but can still change it if you need to). So these changes here are more of a stopgap until we get to that point -- then I'll probably go in and clean up a bunch of this stuff